File: 0078-subsets/0078-subsets.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        ans = []
        p = []
        def rpt(ph, it):
            #print(ph)
            ans.append(list(ph))
            a = list(ph)
            #print(a)
            for i in range(it,len(nums)):
                if nums[i] not in a:
                    a.append(nums[i])
                    print(a)
                    rpt(a, i + 1)
                    a.pop()
        rpt(p, 0)
        return ans
        
        

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:

    void backtrack(vector<int>& nums, vector<vector<int>>& ans, vector<int> hold, int it){
        ans.push_back(hold);

        for(int i = it; i < nums.size(); i++){
            hold.push_back(nums[i]);
            backtrack(nums, ans, hold, i+1);
            hold.pop_back();
        }
    }
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> ans;
        vector<int> hold;
        backtrack(nums, ans, hold, 0);
        return ans;
    }
};